### PR TITLE
Standardised metric naming

### LINKED ISSUE #52: Improve/standardise metric naming + descriptions
Metrics like `solana_confirmed_transactions_total` should ideally be renamed to simply `solana_transactions_total`, as the `confirmed` is misleading given the overlap with solana's commitment levels (current commitment level for this metric is `"finalized"` in any case!).

Additionally, the descriptions on many metrics is rather vague and should be improved!

### PR DESCRIPTION
**Pull request by johnstonematt**
Closes #52

### CODE REVIEW

File: cmd/solana_exporter/exporter.go [hunk 1]
```diff
@@ -39,91 +40,102 @@ type SolanaCollector struct {
...
 	collector := &SolanaCollector{
 		rpcClient:        provider,
 		slotPace:         slotPace,
 		balanceAddresses: CombineUnique(balanceAddresses, nodekeys, votekeys),
 		identity:         identity,
-		totalValidatorsDesc: prometheus.NewDesc(
-			"solana_active_validators",
-			"Total number of active validators by state",
+		ValidatorActive: prometheus.NewDesc(
+			"solana_validator_active",
```
Comment (GranderStark) [inline cmd/solana_exporter/exporter.go:68]: `Solana-Dashboard.json` file  needs to be updated too as it uses that metric

**Comment by GranderStark** [comment]
Just side node as a heads-up for future release notes: That change would break backward compatibility with existing metrics(if someone relies on them).

File: cmd/solana_exporter/exporter.go (hunk 1 above)
Comment (johnstonematt) [reply cmd/solana_exporter/exporter.go:68]: Yeah I know, we want to update the dashboard in general, but thanks for reminding

**Comment by johnstonematt** [comment]
Regarding breaking backwards compatibility - i am aware, but wouldn't changing the labels (which have already been done) also break backwards compatibility?

**Comment by GranderStark** [comment]
> Regarding breaking backwards compatibility - i am aware, but wouldn't changing the labels (which have already been done) also break backwards compatibility?

If we are adding new label to existing metric without changing old label names - that should not entirely break old queries as far as I know. The result may differ depending on how the query is structured and the set of unique label combinations that were used before.

**Comment by johnstonematt** [comment]
> If we are adding new label to existing metric without changing old label names - that shouldn't necessarily break the query's purpose as far as I know. The result may differ depending on how the query is structured and the set of unique label combinations that were used before.

No we explicitly changed the uninformative `pubkey` label to the standardised `votekey` in #43

**Comment by johnstonematt** [comment]
Come to think of it, can't `solana_validator_active` (or `solana_active_validator`, whichever name you want) simply be removed? `solana_validator_delinquent` already tracks whether every validator is delinquent or not, so `solana_validator_active` is just a grouped count of that, no?

@GranderStark

**Comment by GranderStark** [comment]
> > If we are adding new label to existing metric without changing old label names - that shouldn't necessarily break the query's purpose as far as I know. The result may differ depending on how the query is structured and the set of unique label combinations that were used before.
> 
> No we explicitly changed the uninformative `pubkey` label to the standardised `votekey` in #43

Thank you for context, that definitely will cause some impact on those, who filter/aggregate data by old labels 
That change worth mention in release notes/change log

**Comment by GranderStark** [comment]
> Come to think of it, can't `solana_validator_active` (or `solana_active_validator`, whichever name you want) simply be removed? `solana_validator_delinquent` already tracks whether every validator is delinquent or not, so `solana_validator_active` is just a grouped count of that, no?
> 
> @GranderStark

Well, I suppose, yes - `solana_active_validator` just counts length of `voteAccounts.Current` and `voteAccounts.Delinquent`. And both of them in next step is used to populate information in `solana_validator_delinquent`.
Looks like it could be replaced with `count` on the dashboard side.

**Review by SEJeff — APPROVED** [review]

**Pull request merged**